php-cs-fixer

```diff
@@ -37,6 +37,19 @@ if (defined('HHVM_VERSION_ID')) {
     }
 }
 
+foreach (['json', 'tokenizer'] as $extension) {
+    if (!extension_loaded($extension)) {
+        fwrite(STDERR, sprintf("PHP extension ext-%s is missing from your system. Install or enable it.\n", $extension));
+
+        if (getenv('PHP_CS_FIXER_IGNORE_ENV')) {
+            fwrite(STDERR, "Ignoring environment requirements because `PHP_CS_FIXER_IGNORE_ENV` is set. Execution may be unstable.\n");
+        } else {
+            exit(1);
+        }
+    }
+}
+unset($extension);
```

Comment:
You can wrap everything inside a volatile lambda to be sure context gets cleaned at the end:
```php
(static function(){
    $var = 1;
    echo $var;
})();
// $var is undefined now, don't have to remember the unset() call
```

Reply:
I like the idea, though I'd rather do it in a separate PR to have atomic commits as there are other `unset()`s in other places.

Reply:
PHP 5.6 https://3v4l.org/rvgHm :(

Reply:
Definitely later then :)

Though it might be time to drop support for old PHP versions.